CommonTools/PileupAlgos/src/PuppiContainer.cc

```diff
@@ -77,70 +86,51 @@ double PuppiContainer::var_within_R(int iId,
                                     const PuppiCandidate &centre,
                                     const double R) {
   if (iId == -1)
-    return 1;
+    return 1.;
 
-  //this is a circle in rapidity-phi
-  //it would make more sense to have var definition consistent
-  //fastjet::Selector sel = fastjet::SelectorCircle(R);
-  //sel.set_reference(centre);
-  //the original code used Selector infrastructure: it is too heavy here
-  //logic of SelectorCircle is preserved below
+  double const r2(R * R);
+  double var(0.);
 
-  vector<double> near_dR2s;
-  near_dR2s.reserve(std::min(50UL, particles.size()));
-  vector<double> near_pts;
-  near_pts.reserve(std::min(50UL, particles.size()));
-  const double r2 = R * R;
-  for (auto const &part : particles) {
-    if (part.puppi_register() == 3)
-      continue;
-    //squared_distance is in (y,phi) coords: rap() has faster access -> check it first
-    if (std::abs(part.rap() - centre.rap()) < R && part.squared_distance(centre) < r2) {
-      near_dR2s.push_back(reco::deltaR2(part, centre));
-      near_pts.push_back(part.pt());
+  for (auto const &cand : particles) {
+    if (std::abs(cand.rapidity - centre.rapidity) < R) {
+      auto const dr2y(reco::deltaR2(cand.rapidity, cand.phi, centre.rapidity, centre.phi));
+      if (dr2y < r2) {
+        auto const dr2(reco::deltaR2(cand.eta, cand.phi, centre.eta, centre.phi));
```

Comment:
it looks like only ```1./dr2``` is used in a few places below, some speedup should be possible from computing this inversion here